File: app/src/main/java/com/kota/Telnet/TelnetUtils.java
package com.kota.Telnet;

import com.kota.Telnet.Model.TelnetRow;

public class TelnetUtils {
    public static int hashCode(byte[] aData) {
        int hash = 0;
        int multiplier = 1;
        for (int i = aData.length - 1; i >= 0; i--) {
            hash += (aData[i] & 255) * multiplier;
            multiplier = (multiplier << 5) - multiplier;
        }
        return hash;
    }

    public static int getIntegerFromData(TelnetRow aRow, int from, int to) {
        try {
            String temp = aRow.getSpaceString(from, to).trim();
            if (temp.length() > 0) {
                return Integer.parseInt(temp);
            }
            return 0;
        } catch (Exception e) {
            e.printStackTrace();
            return 0;
        }
    }

    public static String getHeader(String source) {
        String trim_source = source.replace(" ", "");
        if (trim_source.length() > 1) {
            return trim_source.substring(0, 2);
        }
        return "";
    }
}

File: app/src/main/java/com/kota/telnet/TelnetUtils.kt
package com.kota.telnet

import android.util.Log
import com.kota.telnet.model.TelnetRow

object TelnetUtils {
    fun hashCode(aData: ByteArray): Int {
        var hash = 0
        var multiplier = 1
        for (i in aData.indices.reversed()) {
            hash += (aData[i].toInt() and 255) * multiplier
            multiplier = (multiplier shl 5) - multiplier
        }
        return hash
    }

    @JvmStatic
    fun getIntegerFromData(aRow: TelnetRow, from: Int, to: Int): Int {
        try {
            val temp = aRow.getSpaceString(from, to).trim { it <= ' ' }
            if (temp.isNotEmpty()) {
                return temp.toInt()
            }
            return 0
        } catch (e: Exception) {
            Log.e(
                TelnetUtils::class.java.simpleName,
                (if (e.message != null) e.message else "")!!
            )
            return 0
        }
    }

    @JvmStatic
    fun getHeader(source: String): String {
        val trimSource = source.replace(" ", "")
        if (trimSource.length > 1) {
            return trimSource.substring(0, 2)
        }
        return ""
    }
}
